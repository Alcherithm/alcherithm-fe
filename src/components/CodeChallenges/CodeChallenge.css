.CodeChallenge {
  padding: 2rem;

  div {
    display: flex;
    justify-content: space-between;
  }
}
