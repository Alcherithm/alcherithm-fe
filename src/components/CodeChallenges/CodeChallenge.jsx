import React from 'react';
import PropTypes from 'prop-types';
import styles from './CodeChallenge.css';

const CodeChallenge = ({ category, challengeNumber, instructions, solutions, char }) => {
  const slicedInstructions = `${instructions.slice(0, char)}...`;
  const icon = solutions[0].passed ? '✔' : '✖️';

  return (
    <section className={styles.CodeChallenge}>
      <h3>{category} {challengeNumber}</h3>
      <p>{slicedInstructions}</p>
      {solutions[0]?.date && <div><h4>{icon}</h4><h4>{solutions[0].date}</h4></div>}
    </section>
  );
};

CodeChallenge.propTypes = {
  category: PropTypes.string.isRequired,
  challengeNumber: PropTypes.number.isRequired,
  instructions: PropTypes.string.isRequired,
  solutions: PropTypes.arrayOf(PropTypes.shape({
    passed: PropTypes.boolean,
    date: PropTypes.string
  })),
  char: PropTypes.number.isRequired
};

export default CodeChallenge;
