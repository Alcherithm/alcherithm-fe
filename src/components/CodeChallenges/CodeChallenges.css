@import '../../styles/variables.css';

.CodeChallenges {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100vw;
  padding-inline-start: 0px;

  h1 {
    text-align: center;
  }

  a {
    text-decoration: none;
    color: var(--gray);
  }

  article {
    display: flex;
    flex-direction: column;
    width: 80%;
  }

  li {
    width: 20rem;
    height: 18rem;
    margin: 2rem;

    list-style-type: none;
    border: 1px solid var(--gray);
    border-radius: 5px;
    box-shadow: var(--box-shadow);  
  }

  li:hover {
    border: 1px solid rgba(49, 232, 232, 0.6);
    box-shadow: var(--hover-box-shadow);
  }

  .row {
    display: flex;
    flex-flow: row wrap;
    justify-content: space-around;
    padding: 3rem;
  }

  .column {
    width: 100%;
    padding-inline-start: 0px;
  }

  .column li {
    display: flex;
    justify-content: flex-start;
    align-items: center;
    width: 100%;
    max-width: 100em;
    min-width: 50em;
    min-height: 15em;
    padding: 2rem;

    section {
      width: 75%;
    }
  }
}
