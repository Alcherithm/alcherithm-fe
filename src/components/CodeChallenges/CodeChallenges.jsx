import React from 'react';
import CodeChallenge from './CodeChallenge';
import { useGetChallenges } from '../../hooks/getChallenges';

const CodeChallenges = () => {
  const challenges = useGetChallenges();
  const codeElements = challenges.map(challenge => (
    <>
      <li key={challenge._id}>
        <CodeChallenge {...challenge} />
      </li>
    </>
  ));

  return (
    <ul>
      {codeElements}
    </ul>
  );
};

export default CodeChallenges;
