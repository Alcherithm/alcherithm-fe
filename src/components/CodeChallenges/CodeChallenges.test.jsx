import React from 'react';
import { shallow } from 'enzyme';
import CodeChallenges from './CodeChallenges';

describe('CodeChallenges component', () => {
  it('renders CodeChallenges', () => {
    const wrapper = shallow(<CodeChallenges />);
    expect(wrapper).toMatchSnapshot();
  });
});
