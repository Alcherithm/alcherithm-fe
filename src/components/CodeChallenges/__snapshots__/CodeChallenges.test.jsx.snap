// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeChallenges component renders CodeChallenges 1`] = `
<section
  className="CodeChallenges"
>
  <Dropdown
    setCategory={[Function]}
  />
  <ul />
</section>
`;
